### PR TITLE
Fix clippy errors in generated files on 1.63 rust

### PR DESCRIPTION
**Pull request by SneakyBerry**
Add derive `Eq` for all types that does not consist nested float type.

### CODE REVIEW

**Comment by lerouxrgd** [comment]
Something like: 

```rs
#[derive(Debug, PartialEq, Eq, Clone)]
pub struct A {
    pub field_one: f32,
}
```

Will not compile since `f32` is not `Eq`.

**Comment by SneakyBerry** [comment]
> Something like:
> 
> ```rust
> #[derive(Debug, PartialEq, Eq, Clone)]
> pub struct A {
>     pub field_one: f32,
> }
> ```
> 
> Will not compile since `f32` is not `Eq`.

Yes, saw it while try to fix tests. Is it ok to use `#[allow(clippy::derive_partial_eq_without_eq)]`?

**Comment by SneakyBerry** [comment]
Or I can try to add condition for add Eq to derives

**Comment by lerouxrgd** [comment]
> Or I can try to add condition for add Eq to derives

It should be doable, I haven't tried it yet because recursively checking whether a record has no f32/f64 is tricky.

**Comment by SneakyBerry** [comment]
> > Or I can try to add condition for add Eq to derives
> 
> It should be doable, I haven't tried it yet because recursively checking whether a record has no f32/f64 is tricky.

Ok. I will try it asap. Thx for hint.

**Comment by SneakyBerry** [comment]
@lerouxrgd I think I made it :)

**Comment by lerouxrgd** [comment]
Thanks for looking into it !

At a first glance I would say there are a few thing than can be simplified:
- Have a private method in `GenState` that takes the `deps: &[Schema]` parameter (which is expected to be calculated by `deps_stack`) and apply your `deep_search_float` logic to produce a `HashSet<String>` of schemas containing a float (where `String` is `serde_json::to_string(schema)`, like in other `GenState` internals).
- Use this method in `GenState::with_deps` to initialize a new field `not_eq: HashSet<String>` in `GenState`
- Add a method in `GenState` like this: `pub fn eq_derivable(&self, schema: &Schema) -> bool` that checks that `schema` is a record or a union, and is not belonging to `not_eq: HashSet<String>`
- Use this new method in `Templater::str_record` and `Templater::str_union_enum` to initialize the value of `ctx.insert("have_float", &have_float);` (maybe rename it `eq_derivable` here too).

File: tests/schemas/nested_with_float.avsc [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (lerouxrgd) [inline tests/schemas/nested_with_float.avsc:1]: Could you try formatting it like the other schemas ?

File: src/gen.rs [hunk 2]
```diff
@@ -132,14 +132,34 @@ impl Generator {
...
+        Schema::Record { fields, .. } => Ok(fields
+            .iter()
+            .any(|f| deep_search_float(&f.schema, deps_map).unwrap())),
+        Schema::Union(union) => Ok(union
+            .variants()
+            .iter()
+            .any(|s| deep_search_float(s, deps_map).unwrap())),
+        Schema::Ref { name } => deps_map
+            .get(name)
+            .ok_or_else(|| Error::Schema("Not found".to_owned()))
```
Comment (martin-g) [inline src/gen.rs:149]: I find it very useful for debugging purposes to have the schema `name` in the error message.

File: tests/schemas/multi_valued_union_records.avsc [hunk 3]
```diff
@@ -1,20 +1,35 @@
 {
   "type": "record",
   "name": "Foo",
-  "fields": [ {
-    "name": "a",
-    "type": "int"
-  }, {
-    "name": "b",
-    "type": [
+  "fields": [
```
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply tests/schemas/multi_valued_union_records.avsc]: Done

File: tests/schemas/nested_with_float.avsc (hunk 1 above)
Comment (SneakyBerry) [reply tests/schemas/nested_with_float.avsc:1]: Is it looks good?

File: src/gen.rs (hunk 2 above)
Comment (SneakyBerry) [reply src/gen.rs]: Done

File: tests/schemas/nested_with_float.avsc (hunk 1 above)
Comment (lerouxrgd) [reply tests/schemas/nested_with_float.avsc:1]: perfect !

**Pull request merged**